base_tier_validation/models/tier_validation.py

```diff
@@ -0,0 +1,595 @@
+# Copyright 2017-19 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from ast import literal_eval
+
+from lxml import etree
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.misc import frozendict
+
+
+class TierValidation(models.AbstractModel):
+    _name = "tier.validation"
+    _description = "Tier Validation (abstract)"
+
+    _tier_validation_buttons_xpath = "/form/header/button[last()]"
+    _tier_validation_manual_config = True
+
+    _state_field = "state"
+    _state_from = ["draft"]
+    _state_to = ["confirmed"]
+    _cancel_state = "cancel"
+
+    # TODO: step by step validation?
+
+    review_ids = fields.One2many(
+        comodel_name="tier.review",
+        inverse_name="res_id",
+        string="Validations",
+        domain=lambda self: [("model", "=", self._name)],
+        auto_join=True,
+    )
+    to_validate_message = fields.Html(compute="_compute_validated_rejected")
+    # TODO: Delete in v17 in favor of validation_status field
```

Comment:
Sorry.. forgot to provide example of those lines..

Reply:
@victoralmau Seems like yoiu've added these comments.
Can you clarify / confirm which fiuelds you are talking about..?
is it about `validated` / `rejected`field?

Reply:
He's on holidays this week. The idea is to have a `validation_status` selection field that unifies the logic for allowing to validate, instead of several boolean fields.